src/training_hub/algorithms/gepa.py

```
@@ -0,0 +1,473 @@
+            algorithm_params: Parameters for the optimization run.
+
+        Returns:
+            GEPAResult from the gepa library.
+        """
+        try:
+            from gepa import optimize
+        except ImportError:
+            raise ImportError(
+                "GEPA requires the 'gepa' package. "
+                "Install it with: pip install training-hub[gepa]"
+            )
```

Ruff (0.15.13)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)